### DISCUSSION TITLE
Common bugs and solutions

### DISCUSSION THREAD

**Discussion by Pelayo-Reguera** [Arquisoft/lomap_en3a#130] [General]
For some time we have been having this issue in which all of our functions are executed twice on our app, and we did not know why, for example when the maps are loaded, (there is a console log to show them), this was appearing on the browser console:
![image](https://user-images.githubusercontent.com/98907563/235144080-cabb045e-6985-4dcc-aa85-7d8d33119591.png)
Meaning this functions where executed twice, in order to solved I have found the following post on StackOverflow shich seems to explain why this was happening: https://stackoverflow.com/questions/50819162/why-is-my-function-being-called-twice-in-react
If we check the React official documentation page: https://react.dev/reference/react/StrictMode we can check that:

> _Strict Mode enables extra development-only checks for the entire component tree inside the <StrictMode> component. These checks help you find common bugs in your components early in the development process._

So as a small conclusion we **do not need to change this** while we are in development, and I think we will not need to touch any of it when we deploy our application, because as it is said in the React documentation: 
> _Although the Strict Mode checks **only run in development**, they help you find bugs that already exist in your code but can be tricky to reliably reproduce in production. Strict Mode lets you fix bugs before your users report them._

### Post made by Guillermo Dylan Carvajal Aza @UO283069

**Comment by GuillermoDylan** [comment]
## Bug description
There seems to be a bug with some unitary tests, it has to do with the **crypto** module that is located inside of the **Place.tsx** class file, this test impedes any react component on our application that has Place imported to be tested, as it always leads to this error.

![image](https://user-images.githubusercontent.com/98907563/235320127-86b5e876-5a8e-42a5-b0b1-b1c21b5b3337.png)

## Solution

Thanks to [this](https://stackoverflow.com/questions/52612122/how-to-use-jest-to-test-functions-using-crypto-or-window-mscrypto) post in Stackoverflow I have found a solution that seems to fix the tests that involve the use of crypto. Adding the following line at the beginning of this tests fix the problem:

```
const crypto = require('crypto');

Object.defineProperty(globalThis, 'crypto', {
    value: {
        randomUUID: () => crypto.randomUUID()
    }
});  
```

### Update
Placing the code in the tests set up ts file has the same effect as placing it on each test and is a quite better soultion.
